File: src/leetcode/0231-Power-of-Two/solution.go
package main

import (
	"fmt"
)

func isPowerOfTwo(n int) bool {
	return n > 0 && (n&-n) == n
}

func main() {
	for _, n := range []int{1, 16, 218} {
		fmt.Println(isPowerOfTwo(n))
	}
}

File: src/leetcode/0231-Power-of-Two/solution.cpp
#include <iostream>

using namespace std;

class Solution {
public:
    bool isPowerOfTwo(int n) { return n > 0 && (n & -n) == n; }
};

int main() {
    Solution solution;
    int inputs[] = {1, 16, 218};
    for (const auto &n : inputs) {
        cout << solution.isPowerOfTwo(n) << endl;
    }
}